### PR TITLE
Add reader::read_bytes_const

### PR DESCRIPTION
**Pull request by wcampbell0x2a**
* Reduce codegen for known sizes at compile time
  - DekuReader<_, Endian> for all types
  - DekuReader<_, (Endian, Endian)> for u8

### CODE REVIEW

File: src/reader.rs [hunk 1]
```diff
@@ -232,6 +233,39 @@ impl<'a, R: Read> Reader<'a, R> {
...
+    ) -> Result<ReaderRet, DekuError> {
+        #[cfg(feature = "logging")]
+        log::trace!("read_bytes: requesting {N} bytes");
+        if self.leftover.is_empty() {
+            if let Err(e) = self.inner.read_exact(buf) {
+                if e.kind() == ErrorKind::UnexpectedEof {
+                    return Err(DekuError::Incomplete(NeedSize::new(N * 8)));
+                }
+
+                // TODO: other errors?
```
Comment (sharksforarms) [inline src/reader.rs:256]: This `read_exact` is off of the `impl std::io::Read` right? So in theory we could get whatever error from the reader (socket error, file io error, etc.)

Is there a reason we don't want to bubble it up?

Comment (wcampbell0x2a) [reply src/reader.rs:256]: Good idea, opened https://github.com/sharksforarms/deku/pull/433 to address this.

Comment (sharksforarms) [reply src/reader.rs:254]: ty, we can now handle this TODO

**Review by sharksforarms — DISMISSED** [review]
Cool!

**Comment by wcampbell0x2a** [comment]
@sharksforarms should be ready now

File: src/reader.rs (hunk 1 above)
Comment (sharksforarms) [inline src/reader.rs:256]: @wcampbell0x2a we could bubble up IO errors here?

Comment (wcampbell0x2a) [reply src/reader.rs:256]: Yep! I won't be back at my computer till later.

Comment (wcampbell0x2a) [reply src/reader.rs]: Ready!

File: src/impls/primitive.rs [hunk 2]
```diff
@@ -79,13 +78,15 @@ impl DekuReader<'_, (Endian, ByteSize)> for u8 {
...
-        let ret = reader.read_bytes(size.0, &mut buf)?;
+        const MAX_TYPE_BYTES: usize = core::mem::size_of::<u8>();
+        let mut buf = [0; MAX_TYPE_BYTES];
+        let ret = reader.read_bytes_const::<MAX_TYPE_BYTES>(&mut buf)?;
         let a = match ret {
             ReaderRet::Bytes => <u8>::from_be_bytes(buf),
             ReaderRet::Bits(bits) => {
                 let Some(bits) = bits else {
-                    return Err(DekuError::Parse("no bits read from reader".to_string()));
+                    // todo!();
```
Comment (sharksforarms) [inline src/impls/primitive.rs:88]: nit: comment

**Review by sharksforarms — DISMISSED** [review]
lgtm, just a nit on the remaining todo comment.

File: src/impls/primitive.rs (hunk 2 above)
Comment (wcampbell0x2a) [reply src/impls/primitive.rs]: updated

**Review by sharksforarms — APPROVED** [review]

**Pull request merged**